File: week-5/n-ary-tree-postorder-traversal/n-ary-tree-postorder-traversal.py
"""
# Definition for a Node.
class Node:
    def __init__(self, val=None, children=None):
        self.val = val
        self.children = children
"""
class Solution:
    def postorder(self, root: 'Node') -> List[int]:
        if not root:
            return []
        res = []
        stack = [root]
        while stack:
            cur = stack.pop();
            for i in cur.children:
                stack.append(i)
            res.append(cur.val)
            
        return res[::-1]

File: week-5/n-ary-tree-postorder-traversal/n-ary-tree-postorder-traversal.js
/**
 * // Definition for a Node.
 * function Node(val,children) {
 *    this.val = val;
 *    this.children = children;
 * };
 */
/**
 * @param {Node} root
 * @return {number[]}
 */
var postorder = function(root) {
    if (!root) return [];
    
    const res = [];
    const stack = [root];
    
    while (stack.length) {
        cur = stack.pop();
        for (let i of cur.children) {
            stack.push(i);
        }
        res.push(cur.val);
    }
    
    return res.reverse();
};
